Alfredo deja de cambiar todos lo modulos, fijate en el travis, al final te muestra que es lo que falla y en este caso es el flake
```
+=======================================
|  Tests summary:
|---------------------------------------
| test_flake8                 FAIL
| test_pylint                 Success
+=======================================
```
Los pylint son warnings, que los ignora segun el archivo de configuración


El error es el siguiente:
```
mrp_repair_partner_lot/models/mrp_repair.py:4:1: F401 'openerp.api' imported but unused
```

--- a/mrp_repair_partner_lot/README.rst
+++ b/mrp_repair_partner_lot/README.rst
@@ -1,0 +1,32 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+
+======================
+Mrp repair partner lot
+======================
+
+* In repair orders new tag "Repair report", with new object
+  "MRP repair customer lot".
+* This new object has the following fields: customer product code, product
+  template, lot, quantity, description breakdown, cause, repair effectuate,
+  repairable.
+* New report "Repair report".
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues
+<https://github.com/avanzosc/mrp-repair-addons/issues>`_. In case of trouble,
+please check there if your issue has already been reported. If you spotted
+it first, help us smash it by providing detailed and welcomed feedback.
+
+Credits
+=======
+
+Contributors
+------------
+* Ana Juaristi <anajuaristi@avanzosc.es>
+* Alfredo de la Fuente <alfredodelafuente@avanzosc.es>
+
+Do not contact contributors directly about support or help with technical issues.

--- a/mrp_repair_partner_lot/__init__.py
+++ b/mrp_repair_partner_lot/__init__.py
@@ -1,0 +1,4 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Alfredo de la Fuente - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from . import models

--- a/mrp_repair_partner_lot/__openerp__.py
+++ b/mrp_repair_partner_lot/__openerp__.py
@@ -1,0 +1,28 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Alfredo de la Fuente - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+{
+    "name": "MRP Repair Partner Lot",
+    "version": "8.0.1.0.0",
+    "license": "AGPL-3",
+    "author": "AvanzOSC",
+    "website": "http://www.avanzosc.es",
+    "contributors": [
+        "Ana Juaristi <anajuaristi@avanzosc.es>",
+        "Alfredo de la Fuente <alfredodelafuente@avanzosc.es>",
+    ],
+    "category": "Manufacturing",
+    "depends": [
+        "stock",
+        "mrp_repair",
+    ],
+    "data": [
+        "security/ir.model.access.csv",
+        "report/mrp_repair_report.xml",
+        "data/report_paperformat.xml",
+        "views/mrp_repair_view.xml",
+        "views/mrp_repair_customer_lot_view.xml",
+    ],
+    "installable": True,
+}

--- a/mrp_repair_partner_lot/data/report_paperformat.xml
+++ b/mrp_repair_partner_lot/data/report_paperformat.xml
@@ -1,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<openerp>
+    <data>
+        <record id="paperformat_mrp_repair_partner_lot" model="report.paperformat">
+            <field name="name">MRP repair partner lot</field>
+            <field name="default" eval="True" />
+            <field name="format">A4</field>
+            <field name="page_height">0</field>
+            <field name="page_width">0</field>
+            <field name="orientation">Landscape</field>
+            <field name="margin_top">30</field>
+            <field name="margin_bottom">10</field>
+            <field name="margin_left">7</field>
+            <field name="margin_right">7</field>
+            <field name="header_line" eval="False" />
+            <field name="header_spacing">25</field>
+            <field name="dpi">90</field>
+        </record>
+        <record id="action_report_mrp_repair_partner_lot" model="ir.actions.report.xml">
+            <field name="paperformat_id" ref="mrp_repair_partner_lot.paperformat_mrp_repair_partner_lot"/>
+        </record>
+    </data>
+</openerp>

--- a/mrp_repair_partner_lot/i18n/es.po
+++ b/mrp_repair_partner_lot/i18n/es.po
@@ -1,0 +1,204 @@
+# Translation of Odoo Server.
+# This file contains the translation of the following modules:
+#	* mrp_repair_partner_lot
+#
+msgid ""
+msgstr ""
+"Project-Id-Version: Odoo Server 8.0\n"
+"Report-Msgid-Bugs-To: \n"
+"POT-Creation-Date: 2019-07-05 09:23+0000\n"
+"PO-Revision-Date: 2019-07-05 09:23+0000\n"
+"Last-Translator: <>\n"
+"Language-Team: \n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: \n"
+"Plural-Forms: \n"
+
+#. module: mrp_repair_partner_lot
+#: view:mrp.repair:mrp_repair_partner_lot.view_repair_order_form_inh_partner_lot
+#: view:mrp.repair.customer.lot:mrp_repair_partner_lot.mrp_repair_customer_lot_view_form
+#: field:mrp.repair.customer.lot,cause:0
+#: view:website:mrp_repair_partner_lot.report_mrp_repair_partner_lot_printing
+msgid "Cause"
+msgstr "Causa"
+
+#. module: mrp_repair_partner_lot
+#: help:mrp.repair.customer.lot,customer_id:0
+msgid "Choose partner for whom the order will be invoiced and delivered."
+msgstr "Elija una empresa para la cual el pedido será facturado y entregado."
+
+#. module: mrp_repair_partner_lot
+#: field:mrp.repair.customer.lot,create_uid:0
+msgid "Created by"
+msgstr "Creado por"
+
+#. module: mrp_repair_partner_lot
+#: field:mrp.repair.customer.lot,create_date:0
+msgid "Created on"
+msgstr "Creado el"
+
+#. module: mrp_repair_partner_lot
+#: view:mrp.repair.customer.lot:mrp_repair_partner_lot.mrp_repair_customer_lot_view_search
+#: field:mrp.repair.customer.lot,customer_id:0
+msgid "Customer"
+msgstr "Cliente"
+
+#. module: mrp_repair_partner_lot
+#: view:mrp.repair:mrp_repair_partner_lot.view_repair_order_form_inh_partner_lot
+#: field:mrp.repair,customer_lot_ids:0
+#: view:mrp.repair.customer.lot:mrp_repair_partner_lot.mrp_repair_customer_lot_view_tree
+msgid "Customer lots"
+msgstr "Lotes cliente"
+
+#. module: mrp_repair_partner_lot
+#: view:mrp.repair.customer.lot:mrp_repair_partner_lot.mrp_repair_customer_lot_view_search
+#: field:mrp.repair.customer.lot,product_code:0
+#: view:website:mrp_repair_partner_lot.report_mrp_repair_partner_lot_printing
+msgid "Customer product code"
+msgstr "Código producto cliente"
+
+#. module: mrp_repair_partner_lot
+#: view:mrp.repair:mrp_repair_partner_lot.view_repair_order_form_inh_partner_lot
+#: view:mrp.repair.customer.lot:mrp_repair_partner_lot.mrp_repair_customer_lot_view_form
+#: field:mrp.repair.customer.lot,description_breakdown:0
+#: view:website:mrp_repair_partner_lot.report_mrp_repair_partner_lot_printing
+msgid "Description breakdown"
+msgstr "Descripción avería"
+
+#. module: mrp_repair_partner_lot
+#: view:mrp.repair.customer.lot:mrp_repair_partner_lot.mrp_repair_customer_lot_view_search
+msgid "Group By"
+msgstr "Agrupar por"
+
+#. module: mrp_repair_partner_lot
+#: field:mrp.repair.customer.lot,id:0
+msgid "ID"
+msgstr "ID"
+
+#. module: mrp_repair_partner_lot
+#: model:ir.model,name:mrp_repair_partner_lot.model_product_supplierinfo
+msgid "Information about a product supplier"
+msgstr "Información de un proveedor de producto"
+
+#. module: mrp_repair_partner_lot
+#: view:website:mrp_repair_partner_lot.report_mrp_repair_partner_lot_printing
+msgid "Invoice address:"
+msgstr "Dirección de facturación:"
+
+#. module: mrp_repair_partner_lot
+#: view:website:mrp_repair_partner_lot.report_mrp_repair_partner_lot_printing
+msgid "Invoice and shipping address:"
+msgstr "Dirección de facturación y de envío:"
+
+#. module: mrp_repair_partner_lot
+#: field:mrp.repair.customer.lot,write_uid:0
+msgid "Last Updated by"
+msgstr "Última modificación por"
+
+#. module: mrp_repair_partner_lot
+#: field:mrp.repair.customer.lot,write_date:0
+msgid "Last Updated on"
+msgstr "Última modificación el"
+
+#. module: mrp_repair_partner_lot
+#: field:mrp.repair.customer.lot,lot_id:0
+#: view:website:mrp_repair_partner_lot.report_mrp_repair_partner_lot_printing
+msgid "Lot"
+msgstr "Lote"
+
+#. module: mrp_repair_partner_lot
+#: view:mrp.repair:mrp_repair_partner_lot.view_repair_order_form_inh_partner_lot
+#: view:mrp.repair.customer.lot:mrp_repair_partner_lot.mrp_repair_customer_lot_view_form
+msgid "MRP repair customer lot"
+msgstr "MRP reparación lote cliente"
+
+#. module: mrp_repair_partner_lot
+#: model:ir.actions.act_window,name:mrp_repair_partner_lot.action_mrp_repair_customer_lot_view
+#: model:ir.ui.menu,name:mrp_repair_partner_lot.mrp_repair_customer_lot_option_view
+msgid "MRP repair customer lots"
+msgstr "MRP reparación lotes cliente"
+
+#. module: mrp_repair_partner_lot
+#: view:mrp.repair:mrp_repair_partner_lot.view_repair_order_form_inh_partner_lot
+msgid "Notes"
+msgstr "Notas"
+
+#. module: mrp_repair_partner_lot
+#: view:mrp.repair.customer.lot:mrp_repair_partner_lot.mrp_repair_customer_lot_view_search
+#: field:mrp.repair.customer.lot,product_tmpl_id:0
+#: view:website:mrp_repair_partner_lot.report_mrp_repair_partner_lot_printing
+msgid "Product template"
+msgstr "Plantilla producto"
+
+#. module: mrp_repair_partner_lot
+#: field:mrp.repair.customer.lot,quantity:0
+#: view:website:mrp_repair_partner_lot.report_mrp_repair_partner_lot_printing
+msgid "Quantity"
+msgstr "Cantidad"
+
+#. module: mrp_repair_partner_lot
+#: view:mrp.repair.customer.lot:mrp_repair_partner_lot.mrp_repair_customer_lot_view_search
+msgid "Repair"
+msgstr "Reparación"
+
+#. module: mrp_repair_partner_lot
+#: model:ir.model,name:mrp_repair_partner_lot.model_mrp_repair
+msgid "Repair Order"
+msgstr "Orden de reparación"
+
+#. module: mrp_repair_partner_lot
+#: view:website:mrp_repair_partner_lot.report_mrp_repair_partner_lot_printing
+msgid "Repair date:"
+msgstr "Fecha reparación:"
+
+#. module: mrp_repair_partner_lot
+#: view:mrp.repair:mrp_repair_partner_lot.view_repair_order_form_inh_partner_lot
+#: view:mrp.repair.customer.lot:mrp_repair_partner_lot.mrp_repair_customer_lot_view_form
+#: field:mrp.repair.customer.lot,repair_made:0
+#: view:website:mrp_repair_partner_lot.report_mrp_repair_partner_lot_printing
+msgid "Repair made"
+msgstr "Reparación efectuada"
+
+#. module: mrp_repair_partner_lot
+#: field:mrp.repair.customer.lot,repair_id:0
+msgid "Repair order"
+msgstr "Orden reparación"
+
+#. module: mrp_repair_partner_lot
+#: model:ir.actions.report.xml,name:mrp_repair_partner_lot.action_report_mrp_repair_partner_lot
+#: view:mrp.repair:mrp_repair_partner_lot.view_repair_order_form_inh_partner_lot
+msgid "Repair report"
+msgstr "Informe reparación"
+
+#. module: mrp_repair_partner_lot
+#: view:website:mrp_repair_partner_lot.report_mrp_repair_partner_lot_printing
+msgid "Repair report N°:"
+msgstr "Informe de reparación N°:"
+
+#. module: mrp_repair_partner_lot
+#: field:mrp.repair.customer.lot,repairable:0
+#: view:website:mrp_repair_partner_lot.report_mrp_repair_partner_lot_printing
+msgid "Repairable"
+msgstr "Reparable"
+
+#. module: mrp_repair_partner_lot
+#: view:website:mrp_repair_partner_lot.report_mrp_repair_partner_lot_printing
+msgid "Responsible:"
+msgstr "Responsable:"
+
+#. module: mrp_repair_partner_lot
+#: view:mrp.repair.customer.lot:mrp_repair_partner_lot.mrp_repair_customer_lot_view_search
+msgid "Search mrp repair customer lots"
+msgstr "Buscar lotes cliente reparación"
+
+#. module: mrp_repair_partner_lot
+#: view:website:mrp_repair_partner_lot.report_mrp_repair_partner_lot_printing
+msgid "Shipping address :"
+msgstr "Dirección de envío :"
+
+#. module: mrp_repair_partner_lot
+#: view:website:mrp_repair_partner_lot.report_mrp_repair_partner_lot_printing
+msgid "VAT:"
+msgstr "NIF:"
+

--- a/mrp_repair_partner_lot/models/__init__.py
+++ b/mrp_repair_partner_lot/models/__init__.py
@@ -1,0 +1,5 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Alfredo de la Fuente - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from . import mrp_repair
+from . import product_supplierinfo

--- a/mrp_repair_partner_lot/models/mrp_repair.py
+++ b/mrp_repair_partner_lot/models/mrp_repair.py
@@ -1,0 +1,37 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Alfredo de la Fuente - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from openerp import models, fields
+import openerp.addons.decimal_precision as dp
+
+
+class MrpRepair(models.Model):
+    _inherit = 'mrp.repair'
+
+    customer_lot_ids = fields.One2many(
+        comodel_name='mrp.repair.customer.lot',
+        inverse_name='repair_id', string='Customer lots')
+
+
+class MrpRepairCustomerLot(models.Model):
+    _name = 'mrp.repair.customer.lot'
+    _description = "MRP repair customer lot"
+
+    repair_id = fields.Many2one(
+        comodel_name='mrp.repair', string='Repair order')
+    customer_id = fields.Many2one(
+        comodel_name='res.partner', string='Customer',
+        related='repair_id.partner_id', store=True)
+    product_code = fields.Many2one(
+        string='Customer product code', comodel_name='product.supplierinfo')
+    product_tmpl_id = fields.Many2one(
+        comodel_name='product.template', string='Product template',
+        related='product_code.product_tmpl_id', store=True)
+    lot_id = fields.Many2one(
+        comodel_name='stock.production.lot', string='Lot')
+    quantity = fields.Float(
+        string='Quantity', digits=dp.get_precision('Product Unit of Measure'))
+    description_breakdown = fields.Text(string='Description breakdown')
+    cause = fields.Text(string='Cause')
+    repair_made = fields.Text(string='Repair made')
+    repairable = fields.Boolean(string='Repairable')

--- a/mrp_repair_partner_lot/models/product_supplierinfo.py
+++ b/mrp_repair_partner_lot/models/product_supplierinfo.py
@@ -1,0 +1,21 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Alfredo de la Fuente - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from openerp import models, api
+
+
+class ProductSupplierinfo(models.Model):
+    _inherit = 'product.supplierinfo'
+
+    @api.multi
+    def name_get(self):
+        if self.env.context.get('show_customer_product_code', False):
+            new_res = []
+            for product in self.filtered(
+                    lambda x: x.product_code or x.product_name):
+                name = u"[{}] {}".format(
+                    product.product_code or '', product.product_name or '')
+                new_res.append((product.id, name))
+            if new_res:
+                return new_res
+        return super(ProductSupplierinfo, self).name_get()

--- a/mrp_repair_partner_lot/report/mrp_repair_report.xml
+++ b/mrp_repair_partner_lot/report/mrp_repair_report.xml
@@ -1,0 +1,124 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+    <data>
+        <report
+            string="Repair report"
+            id="action_report_mrp_repair_partner_lot"
+            model="mrp.repair"
+            report_type="qweb-pdf"
+            name="mrp_repair_partner_lot.report_mrp_repair_partner_lot"
+            file="mrp_repair_partner_lot.report_mrp_repair_partner_lot"/>
+
+        <template id="report_mrp_repair_partner_lot_printing">
+            <t t-call="report.external_layout">
+                <div class="page">
+                    <div class="oe_structure"/>
+                    <div class="row">
+                        <div class="col-xs-6">
+                            <strong t-if="o.address_id == o.partner_invoice_id">Invoice and shipping address:</strong>
+                            <div t-if="o.partner_invoice_id">
+                                <strong t-if="o.address_id != o.partner_invoice_id">Invoice address: </strong>
+                                <div t-field="o.partner_invoice_id"
+                                    t-field-options='{"widget": "contact", "fields": ["address", "name", "phone", "fax"], "no_marker": true}'/>
+                                <p t-if="o.partner_id.vat">VAT: <span t-field="o.partner_id.vat"/></p>
+                            </div>
+                            <div t-if="o.address_id != o.partner_invoice_id" class="mt8">
+                                <strong>Shipping address :</strong>
+                                <div t-field="o.address_id" 
+                                    t-field-options='{"widget": "contact", "fields": ["address", "name", "phone", "fax"], "no_marker": true}'/>
+                                <p t-if="o.partner_id.vat">VAT: <span t-field="o.partner_id.vat"/></p>
+                            </div>                        
+                        </div>
+                        <div class="col-xs-5 col-xs-offset-1">
+                            <div t-field="o.partner_id"
+                                t-field-options='{"widget": "contact", "fields": ["address", "name"], "no_marker": true}' />
+                        </div>
+                    </div>
+
+                    <h2>
+                        <span>Repair report N°:</span>
+                        <span t-field="o.name"/>
+                    </h2>
+
+                    <div class="row mt32 mb32">
+                        <div class="col-xs-3">
+                            <strong>Repair date:</strong>
+                            <br/>
+                            <span t-field="o.date_repair" t-field-options='{"widget": "date"}'/>
+                        </div>
+                        <div t-if="o.user_id.name" class="col-xs-3">
+                            <strong>Responsible:</strong>
+                            <p t-field="o.user_id.name"/>
+                        </div>
+                    </div>
+
+                    <table class="table table-condensed">
+                        <thead>
+                            <tr name="report-head">
+                                <th>Customer product code</th>
+                                <th>Product template</th>
+                                <th>Lot</th>
+                                <th>Quantity</th>
+                                <th>Description breakdown</th>
+                                <th>Cause</th>
+                                <th>Repair effectuate</th>
+                                <th>Repairable</th>
+                            </tr>
+                        </thead>
+                        <tbody>
+                            <tr t-foreach="o.customer_lot_ids" t-as="line" name="report-body">
+                                <td>
+                                    <p t-if="line.product_code">
+                                       <span t-field="line.product_code" />
+                                     </p>
+                                </td>
+                                <td>
+                                    <p t-if="line.product_tmpl_id">
+                                       <span t-field="line.product_tmpl_id" />
+                                     </p>
+                                </td>
+                                <td>
+                                    <p t-if="line.lot_id">
+                                       <span t-field="line.lot_id" />
+                                     </p>
+                                </td>
+                                <td>
+                                    <p t-if="line.quantity">
+                                       <span t-field="line.quantity" />
+                                     </p>
+                                </td>
+                                <td>
+                                    <p t-if="line.description_breakdown">
+                                       <span t-field="line.description_breakdown" />
+                                     </p>
+                                </td>
+                                <td>
+                                    <p t-if="line.cause">
+                                       <span t-field="line.cause" />
+                                     </p>
+                                </td>
+                                <td>
+                                    <p t-if="line.repair_made">
+                                       <span t-field="line.repair_made" />
+                                     </p>
+                                </td>
+                                <td>
+                                    <span t-field="line.repairable" />
+                                </td>
+                            </tr>
+                        </tbody>
+                    </table>
+                    <div class="oe_structure"/>
+                </div>
+            </t>
+        </template>
+
+        <template id="report_mrp_repair_partner_lot">
+            <t t-call="report.external_layout">
+                <t t-foreach="doc_ids" t-as="doc_id">
+                    <t t-raw="translate_doc(doc_id, doc_model, 'partner_id.lang', 'mrp_repair_partner_lot.report_mrp_repair_partner_lot_printing')"/>
+                </t>
+            </t>
+        </template>
+    </data>
+</openerp>

--- a/mrp_repair_partner_lot/security/ir.model.access.csv
+++ b/mrp_repair_partner_lot/security/ir.model.access.csv
@@ -1,0 +1,4 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+access_mrp_repair_customer_lot_manager,mrp.repair.customer.lot,model_mrp_repair_customer_lot,mrp.group_mrp_manager,1,1,1,1
+access_mrp_repair_customer_lot_user,mrp.repair.customer.lot,model_mrp_repair_customer_lot,mrp.group_mrp_user,1,1,1,1
+

--- a/mrp_repair_partner_lot/tests/__init__.py
+++ b/mrp_repair_partner_lot/tests/__init__.py
@@ -1,0 +1,4 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Alfredo de la Fuente - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from . import test_mrp_repair_partner_lot

--- a/mrp_repair_partner_lot/tests/test_mrp_repair_partner_lot.py
+++ b/mrp_repair_partner_lot/tests/test_mrp_repair_partner_lot.py
@@ -1,0 +1,29 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Alfredo de la Fuente - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+import openerp.tests.common as common
+
+
+class TestMrpRepairPartnerLot(common.TransactionCase):
+
+    def setUp(self):
+        super(TestMrpRepairPartnerLot, self).setUp()
+        self.supplierinfo = self.env['product.supplierinfo']. search(
+            [], limit=1)
+        self.supplierinfo.write(
+            {'product_code': 'a1a1a1',
+             'product_name': 'AAAAAA'})
+
+    def test_mrp_repair_partner_lot(self):
+        res = self.supplierinfo.name_get()
+        self.assertEqual(
+            str(res),
+            "[({}, u'{}')]".format(
+                self.supplierinfo.id, self.supplierinfo.name.name))
+        res = self.supplierinfo.with_context(
+            show_customer_product_code=True).name_get()
+        self.assertEqual(
+            str(res),
+            "[({}, u'[{}] {}')]".format(
+                self.supplierinfo.id, self.supplierinfo.product_code,
+                self.supplierinfo.product_name))

--- a/mrp_repair_partner_lot/views/mrp_repair_customer_lot_view.xml
+++ b/mrp_repair_partner_lot/views/mrp_repair_customer_lot_view.xml
@@ -1,0 +1,89 @@
+<?xml version="1.0"?>
+<openerp>
+    <data>
+        <record id="mrp_repair_customer_lot_view_tree" model="ir.ui.view">
+            <field name="name">mrp.repair.customer.lot.view.tree</field>
+            <field name="model">mrp.repair.customer.lot</field>
+            <field name="arch" type="xml">
+                 <tree string="Customer lots">
+                     <field name="repair_id" />
+                     <field name="customer_id" />
+                     <field name="product_code" context="{'show_customer_product_code':True}"/>
+                     <field name="product_tmpl_id" />
+                     <field name="lot_id" />
+                     <field name="quantity" />
+                     <field name="description_breakdown" />
+                     <field name="cause" />
+                     <field name="repair_made" />
+                     <field name="repairable" />
+                 </tree>
+            </field>
+        </record>
+        <record id="mrp_repair_customer_lot_view_search" model="ir.ui.view">
+            <field name="name">mrp.repair.customer.lot.view.search</field>
+            <field name="model">mrp.repair.customer.lot</field>
+            <field name="arch" type="xml">
+                <search string="Search mrp repair customer lots">
+                    <field name="repair_id" />
+                    <field name="customer_id" />
+                    <field name="product_code" />
+                    <field name="product_tmpl_id" />
+                    <field name="lot_id" />
+                    <group expand="1" string="Group By">
+                        <filter string="Customer" name="group-customer"
+                                domain="[]" context="{'group_by':'customer_id'}"/>
+                        <filter string="Repair" name="group-repair"
+                                domain="[]" context="{'group_by':'repair_id'}"/>
+                        <filter string="Customer product code" name="group-product-code"
+                                domain="[]" context="{'group_by':'product_code'}"/>
+                        <filter string="Product template" name="group-product-template"
+                                domain="[]" context="{'group_by':'product_tmpl_id'}"/>
+                    </group>
+                </search>
+            </field>
+        </record>
+
+        <record id="mrp_repair_customer_lot_view_form" model="ir.ui.view">
+            <field name="name">mrp.repair.customer.lot.view.form</field>
+            <field name="model">mrp.repair.customer.lot</field>
+            <field name="arch" type="xml">
+                <form string="MRP repair customer lot">
+                    <sheet>
+                        <group name="head">
+                            <group>
+                                <field name="repair_id" />
+                                <field name="customer_id" invisible="1"/>
+                                <field name="product_code"
+                                    domain="[('name', '=', customer_id), ('type', '=', 'customer')]"
+                                    context="{'show_customer_product_code':True}" />
+                                <field name="product_tmpl_id" readonly="1"/>
+                            </group>
+                            <group>
+                                <field name="lot_id" />
+                                <field name="quantity" />
+                                <field name="repairable" />
+                            </group>
+                            <separator string="Description breakdown" colspan="4" />
+                            <field name="description_breakdown" nolabel="1" colspan="4" />
+                            <separator string="Cause" colspan="4" />
+                            <field name="cause" nolabel="1" colspan="4" />
+                            <separator string="Repair made" colspan="4" />
+                            <field name="repair_made" nolabel="1" colspan="4" />
+                        </group>
+                    </sheet>
+                </form>
+            </field>
+        </record>
+        <record id="action_mrp_repair_customer_lot_view" model="ir.actions.act_window">
+            <field name="name">MRP repair customer lots</field>
+            <field name="type">ir.actions.act_window</field>
+            <field name="res_model">mrp.repair.customer.lot</field>
+            <field name="view_type">form</field>
+            <field name="view_mode">tree,form</field>
+        </record>
+        <menuitem id="mrp_repair_customer_lot_option_view" name="MRP repair customer lots"
+                  action="action_mrp_repair_customer_lot_view" 
+                  groups="mrp.group_mrp_user"
+                  parent="mrp.menu_mrp_manufacturing" sequence="51"/>
+    </data>
+</openerp>

--- a/mrp_repair_partner_lot/views/mrp_repair_view.xml
+++ b/mrp_repair_partner_lot/views/mrp_repair_view.xml
@@ -1,0 +1,58 @@
+<?xml version="1.0"?>
+<openerp>
+    <data>
+        <record id="view_repair_order_form_inh_partner_lot" model="ir.ui.view">
+            <field name="name">view.repair.order.form.inh.partner.lot</field>
+            <field name="model">mrp.repair</field>
+            <field name="inherit_id" ref="mrp_repair.view_repair_order_form"/>
+            <field name="arch" type="xml">
+                <page string="Notes" position="after">
+                    <page string="Repair report" name="repair-report">
+                        <group colspan="4">
+                            <field name="customer_lot_ids" nolabel="1" 
+                                    context="{'default_repair_id': active_id, 'show_customer_product_code':True}">
+                                <tree string="Customer lots" name="tree-customer-lots">
+                                    <field name="repair_id" invisible="1"/>
+                                    <field name="customer_id" invisible="1" />
+                                    <field name="product_code" />
+                                    <field name="product_tmpl_id" />
+                                    <field name="lot_id" />
+                                    <field name="quantity" />
+                                    <field name="description_breakdown" />
+                                    <field name="cause" />
+                                    <field name="repair_made" />
+                                    <field name="repairable" />
+                                </tree>
+                                <form string="MRP repair customer lot" name="form-customer-lots">
+                                    <sheet>
+                                        <group name="head">
+                                            <group>
+                                                <field name="repair_id" invisible="1"/>
+                                                <field name="customer_id" invisible="1"/>
+                                                <field name="product_code"
+                                                    domain="[('name', '=', customer_id), ('type', '=', 'customer')]"
+                                                    context="{'show_customer_product_code':True}" />
+                                                <field name="product_tmpl_id" readonly="1" />
+                                            </group>
+                                            <group>
+                                                <field name="lot_id" />
+                                                <field name="quantity" />
+                                                <field name="repairable" />
+                                            </group>
+                                            <separator string="Description breakdown" colspan="4" />
+                                            <field name="description_breakdown" nolabel="1" colspan="4" />
+                                            <separator string="Cause" colspan="4" />
+                                            <field name="cause" nolabel="1" colspan="4" />
+                                            <separator string="Repair made" colspan="4" />
+                                            <field name="repair_made" nolabel="1" colspan="4" />
+                                        </group>
+                                    </sheet>
+                                </form>
+                            </field>
+                        </group>
+                    </page>
+                </page>
+            </field>
+        </record>
+    </data>
+</openerp>

